sympy/stats/drv_types.py

```diff
@@ -26,6 +28,20 @@ def check(lamda):
     def pdf(self, k):
         return self.lamda**k / factorial(k) * exp(-self.lamda)
 
+    def sample(self):
+        u = random.uniform(0, 1)
+        inf = self.set.inf
+        interval = Interval(self.cdf(inf), self.cdf(inf + 1))
```

Comment:
Since you mentioned efficiency, I wonder if the overhead of creating these Interval objects is worth it. It seems your code amounts to the following:
```
u = random.uniform(0, 1)
inf = self.set.inf
while u > self.cdf(inf):
    inf += 1
return inf
```
I think this version puts the meaning across more clearly: we return the smallest number of the form (inf + integer) where cdf is <= u.